geopandas/array.py

```diff
@@ -13,9 +13,14 @@
 import shapely.ops
 import shapely.wkt
 
-from collections.abc import Iterable
+try:
+    import pygeos
+except ImportError:
+    geos = None
 
 from ._compat import PANDAS_GE_024, PANDAS_GE_10
+from . import _compat as compat
```

Comment:
Yeah, the reason I needed it here as compat.USE_PYGEOS is that I enabled you to change this after importing, which needs to access the variable from the module to have it always point to the latest value. 
But agree consistency is better, so will change the pandas ones to do the same.

Reply:
My 2 cents: I really like the pattern of `compat.USE_PYGEOS` actually. I feel like all of these checks for optional dependencies and versions should be moved to `compat`, for example, we could have `compat.HAS_RTREE` and/or `compat.HAS_SINDEX`. And then all modules should just import `compat` (like is being done here) so that things can be dynamically changed.

I also think it would be good to have separate flags for `HAS_XYZ` and `USE_XYZ`, so that we can do something like this:
```python3
from . import _compat as compat


if compat.HAS_PYGEOS:
    import pygeos
    class A(pygeos....):
    ...
```

Which is maybe a better pattern than [this](https://github.com/geopandas/geopandas/blob/a6463926b09fde703e246a936bba4e58025a0fec/geopandas/base.py#L22) (I could be wrong about this being better, just my thoughts).